### PR TITLE
Model Validation Tests

### PR DESCRIPTION
**Pull request by Chimoneg27**
### For this feature I did the following

- Removed linter errors
- Used git flow
- Created the tests for model methods
- Created tests for model validations
- Updated gemlock file

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@ class Post < ApplicationRecord
   has_many :likes, foreign_key: 'post_id'
   has_many :comments, foreign_key: 'post_id'
 
+  validates :title, presence: true, length: { maximum: 255 }
```
Comment (zeff96) [inline app/models/post.rb]: - [ ] Hey there! 🌟 Great job on adding validation for the post title! 🎉 However, there's a tiny tweak needed to align with the project requirements. The maximum character limit for the title should be 250, but in the current implementation, it's set to 255. 🚨 Let's adjust that to ensure we're spot-on with the project specifications! 🛠️ Thanks a bunch! 👍

Screenshot from 2023-12-01 20-03-36

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  it 'is valid with a title and other attributes' do
+    post = FactoryBot.build(:post, author_id: user.id)
+    expect(post).to be_valid
+  end
+
+  it 'is not valid with more than 255 characters' do
+    post = FactoryBot.build(:post, title: 'a' * 256, author_id: user.id)
+    expect(post).not_to be_valid
+  end
```
Comment (zeff96) [inline spec/models/post_spec.rb]: - [ ] Hey! 👋 I noticed a slight misalignment with the project requirements in our testing approach. 🧐 While the project clearly states that the post title should not exceed 250 characters, our test asserts that a title of 256 characters is invalid. 🚫 Let's ensure our test accurately reflects the project specs by adjusting the assertion to check for a maximum of 250 characters. 📏 This way, our tests align perfectly with the project requirements! 🌟 Thanks for your attention to detail! 🙌

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️
Hi @Chimoneg27 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
:heavy_check_mark: Validations added
:heavy_check_mark: Unit tests added



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @Chimoneg27, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 



## Highlights
- Implemented the features as per the previous code reviewer's suggested changes ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**